ironfish/src/migrations/data/013-wallet-2.ts

```diff
@@ -0,0 +1,354 @@
+    logger.debug('\t' + BenchUtils.renderSegment(BenchUtils.endSegment(start)))
+
+    logger.debug('Migrating: meta')
+    start = BenchUtils.startSegment()
+    await this.migrateMeta(stores.old.meta, stores.new.meta, accounts, tx, logger)
+    logger.debug('\t' + BenchUtils.renderSegment(BenchUtils.endSegment(start)))
+
+    logger.debug('Migrating: Deleting old store nullifierToNote')
+    start = BenchUtils.startSegment()
+    await stores.old.nullifierToNote.clear(tx)
+    logger.debug('\t' + BenchUtils.renderSegment(BenchUtils.endSegment(start)))
+
+    logger.debug('Migrating: Deleting old store noteToNullifier')
+    start = BenchUtils.startSegment()
+    await stores.old.noteToNullifier.clear(tx)
+    logger.debug('\t' + BenchUtils.renderSegment(BenchUtils.endSegment(start)))
+
+    logger.debug(BenchUtils.renderSegment(BenchUtils.endSegment(startTotal)))
+  }
+
+  backward(): Promise<void> {
+    throw new Error()
+  }
+
+  async buildNoteToTransactionCache(
+    transactions: Stores['old']['transactions'],
+    tx: IDatabaseTransaction,
+    logger: Logger,
+  ): Promise<BufferMap<Buffer>> {
+    const noteToTransaction = new BufferMap<Buffer>()
+    let tranactionCount = 0
+
+    for await (const [transactionHash, transactionEntry] of transactions.getAllIter(tx)) {
+      const transaction = new Transaction(transactionEntry.transaction)
+
+      for (const note of transaction.notes()) {
+        const noteHash = note.merkleHash()
+        noteToTransaction.set(noteHash, transactionHash)
+      }
+
+      tranactionCount++
+    }
+
+    logger.debug(
+      `\tFound ${noteToTransaction.size} notes that map to ${tranactionCount} transactions`,
+    )
+
+    return noteToTransaction
+  }
+
+  async migrateBalances(
+    balancesStoreNew: Stores['new']['balances'],
+    unconfirmedBalances: Map<string, bigint>,
+    accounts: { account: DatabaseStoreValue<NewStores['accounts']>; id: string }[],
+    tx: IDatabaseTransaction,
+    logger: Logger,
+  ): Promise<void> {
+    for (const account of accounts) {
+      const balance = unconfirmedBalances.get(account.id)
+
+      if (typeof balance === 'bigint') {
+        logger.debug(`\tCalculated balance ${account.account.name}: ${balance}`)
+        await balancesStoreNew.put(account.id, balance, tx)
+      } else {
+        logger.debug(`\tNo balance for ${account.account.name}, setting to 0`)
+        await balancesStoreNew.put(account.id, BigInt(0), tx)
+      }
+    }
+  }
+
+  async migrateNullifierToNoteHash(
+    nullifierToNoteOld: Stores['old']['nullifierToNote'],
+    nullifierToNoteHashNew: Stores['new']['nullifierToNoteHash'],
+    tx: IDatabaseTransaction,
+    logger: Logger,
+  ): Promise<void> {
+    let count = 0
+
+    for await (const [nullifier, noteHash] of nullifierToNoteOld.getAllIter(tx)) {
+      logger.debug(`\tMigrating note's nullifier: ${HashUtils.renderHashHex(noteHash)}`)
```

Comment:
If we remove this line, the log output is pretty concise. It's a shame we don't have 1 extra level of verbosity, would be great for stuff like this. Curious what you think about removing this one, I understand this probably has/had some use, so if we want to keep it, that's cool too.
```suggestion
```

Log output is fixed size, even with >100k notes 
![image](https://user-images.githubusercontent.com/97762857/182905784-d2cb8de7-d067-4956-a1a5-500e20dd587d.png)